### PR TITLE
feat: decompress json section if not url

### PR DESCRIPTION
**Pull request by willemneal**
Uses brotli to decompress the json data.

### CODE REVIEW

File: craco.config.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+//   //     console.log("!hello!")
+//   //     const experiments = config.experiments || {};
+//   //     config.experiments = {
+//   //       ...experiments,
+//   //       syncWebAssembly: true,
+//   //       asyncWebAssembly: true,
+//   //     };
+//   //     return config
+//   //   }
+// };
```
Comment (chadoh) [inline craco.config.js]: Lots of comments in this file. Are they useful, or should we delete them?

File: src/near/schema.ts [hunk 2]
```diff
@@ -4,42 +4,64 @@ import { readCustomSection } from "wasm-walrus-tools"
...
+    return schema;
+  }
 
   // TODO validate schema adheres to JSONSchema7
-  return schema
+  return JSON.parse(url_or_data)
 }
 
-class NoCustomSectionError extends Error {
+class NoCustomSection extends Error {
```
Comment (chadoh) [inline src/near/schema.ts:26]: rude!

File: src/near/schema.ts [hunk 3]
```diff
@@ -4,42 +4,64 @@ import { readCustomSection } from "wasm-walrus-tools"
 import { ContractCodeView } from "near-api-js/lib/providers/provider"
 import { JSONSchema7 } from "json-schema"
 
+
 export async function fetchSchema(contract: string, near: naj.Near): Promise<JSONSchema7> {
   // TODO handle either HTTP endpoint or IPFS hash
-  const url = await fetchJsonAddress(contract, near)
+  const url_or_data = await fetchJsonAddressOrData(contract, near)
```
Comment (chadoh) [inline src/near/schema.ts]: you're working with TS now 😉 

```suggestion
  const urlOrData = await fetchJsonAddressOrData(contract, near)
```

File: src/near/schema.ts [hunk 4]
```diff
@@ -4,42 +4,64 @@ import { readCustomSection } from "wasm-walrus-tools"
...
+
 
   // TODO cache schema JSON in localeStorage, return early here if available
 
-  const schema = fetch(url)
+  if (url_or_data.startsWith("https://")) {
```
Comment (chadoh) [inline src/near/schema.ts]:
```suggestion
  if (urlOrData.startsWith("https://")) {
```

File: src/near/schema.ts [hunk 5]
```diff
@@ -4,42 +4,64 @@ import { readCustomSection } from "wasm-walrus-tools"
...
     .then((response) => {
       if (!response.ok) throw new Error(`HTTP error! Status: ${response.status}`)
       return response.json()
     })
+    return schema;
+  }
 
   // TODO validate schema adheres to JSONSchema7
-  return schema
+  return JSON.parse(url_or_data)
```
Comment (chadoh) [inline src/near/schema.ts]:
```suggestion
  return JSON.parse(urlOrData)
```

**Review by chadoh — CHANGES_REQUESTED** [review]
I think it probably makes sense to only support one approach for now. Go all-in on putting the AJV in a Custom Section for now. If people want the URL approach to be added again later, we can add support for it later. No need to incur extra maintenance cost for it while we're still figuring things out.

File: src/near/schema.ts (hunk 2 above)
Comment (chadoh) [reply src/near/schema.ts:26]: (sort of joking; I don't really care what we call this variable. But I do find the rename to be a pointless change that adds noise to this PR)

Comment (willemneal) [reply src/near/schema.ts:26]: Fair enough, just a pet peeve of mine.  `throw new NoCustomSection` is clearly an error.  Can remove if you find it too noisy.

File: src/near/schema.ts (hunk 3 above)
Comment (willemneal) [reply src/near/schema.ts]: too much rust...

File: craco.config.js (hunk 1 above)
Comment (willemneal) [reply craco.config.js]: They were passed failed attempts.  I'll delete them.

File: src/near/schema.ts (hunk 2 above)
Comment (chadoh) [reply src/near/schema.ts:26]: I'll let it slide 🙃

**Review by chadoh — APPROVED** [review]

**Pull request merged**